pinot-core/src/main/java/org/apache/pinot/core/realtime/stream/StreamMessageDecoder.java

```diff
@@ -58,4 +58,40 @@ void init(Map<String, String> props, Schema indexingSchema, String topicName)
    * @return A new row decoded from the buffer
    */
   GenericRow decode(T payload, int offset, int length, GenericRow destination);
+
```

Comment:
should we deprecate the methods in lines 53 and  60?

Reply:
I will add deprecations in a followup as it needs to be done across the interfaces we have touched here.